src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardIdentification.cs

```diff
@@ -0,0 +1,337 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Identification of Raspberry Pi board models
+    /// </summary>
+    internal class RaspberryBoardIdentification
+    {
+        /// <summary>
+        /// The Raspberry Pi model.
+        /// </summary>
+        public enum Model
+        {
+            /// <summary>
+            /// Unknown model.
+            /// </summary>
+            Unknown,
```

Comment:
Currently this is not an issue but I suspect in the future this will become a bit problematic as adding new entry which is a variant of existing one will require updating all places which use the branching. This is fine for now since this is not an issue yet and code is internal but we should think of how to express these kinds of relationship.

Reply:
Yea, true. Outside of the actual identification, I think the best way will be to use inheritance for the class features, (similar to the GpioDriver tree, but on a more generic level).

Reply:
this shouldn't be necessary, I'd recommend just to construct a tree/graph (static field with `Dictionary<Model, Model>` with all related entries) which would describe how the things are related and then add a method to have a nicer way to get the relationship (i.e. you can describe who is parent of what, i.e. RaspberryPi3 is parent of RaspberryPi3A, RaspberryPi3B and then you can create method which would tell you if any of the values is RaspberryPi3), you can define even multilevel relationships. I'd start with brainstorming first rather than fixing it right now